timdex_dataset_api/run.py

```diff
@@ -3,25 +3,19 @@
 import concurrent.futures
 import logging
 import time
-from typing import TYPE_CHECKING
 
 import pandas as pd
+import pyarrow.dataset as ds
 import pyarrow.parquet as pq
 
-if TYPE_CHECKING:
-    from timdex_dataset_api.dataset import TIMDEXDataset
-
 logger = logging.getLogger(__name__)
 
 
 class TIMDEXRunManager:
     """Manages and provides access to ETL run metadata from the TIMDEX parquet dataset."""
 
-    def __init__(self, timdex_dataset: "TIMDEXDataset"):
-        self.timdex_dataset: TIMDEXDataset = timdex_dataset
-        if self.timdex_dataset.dataset is None:
-            self.timdex_dataset.load()
-
+    def __init__(self, dataset: ds.Dataset):
+        self.dataset = dataset
```

Comment:
I meant to break this into a distinct commit, but this is just slightly reworking `TIMDEXRunManager` to accept a `pyarrow.Dataset` instance versus a `TIMDEXDataset` instance to reduce an otherwise tight coupling.